### PR TITLE
Fixes items preventing other switch in effects

### LINKED ISSUE #5427: Switch in ability bug when mulitply mons faint at the same time
### Description

Ya know I was originally submitting this for Booster Energy/Protosynthesis not procing when it was supposed to, but I'm starting to think there's a more general "things that happen on throw in" issue, maybe related to #5425 as well.

Here's a vid that has a few bugs in it, taken on master. The bugs:
Roaring moon's protosynthesis gets activated by its booster energy, but the ability popup shows up in the wrong place and shows the wrong ability.
Torkoal's drought does not activate.
Gouging Fire's protosynthesis does not activate, it is holding a booster energy as well.
I was originally trying to capture what i thought was a bug if 2 mons with booster energies/protosynthesis were thrown in, only one would proc, but there's a lot goin on here.

https://github.com/user-attachments/assets/11cfd852-6230-4b15-b743-d29764268b46

pokeemerald-1
pokeemerald-3
pokeemerald-4

### Version

1.9.2 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

iriv24

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #5427

Despite the description of the issue, this was an issue with items. After a switch in when items are triggered they blocked all other effect from happening. The problem was the place they were invoked from. To fix this I Introduced `ITEMEFFECT_ON_SWITCH_IN_FIRST_TURN` which will only work for first turn effects and `ITEMEFFECT_ON_SWITCH_IN` will be used for the rest. By passing the caseID as the argument we can easily switch between the scripts we want to run. 

Tests were provided by @PhallenTree

### CODE REVIEW

**Review by Pawkkie — DISMISSED** [review]
I think this is a *massive* improvement to readability along with the bugfix. Approving but not merging in case you'd like another battle-script-centric reviewer to give it a once over first, though feel free to just merge when you wake up :)

**Comment by AlexOn1ine** [comment]
Sorry @Pawkkie, I found an other minor optimization to do so dismissed the review. Feel free to approve and merge again.

**Comment by Pawkkie** [comment]
Okay got it sorted properly this time, conflicts should be fixed, let's hope CLI agrees lol

**Review by Pawkkie — APPROVED** [review]

**Pull request merged**
